message/ipldbind/schema.ipldsch

```diff
@@ -2,36 +2,52 @@ type GraphSyncExtensions {String:nullable Any}
 type GraphSyncRequestID bytes
 type GraphSyncPriority int
 
+type GraphSyncLinkAction enum {
+   # Present means the linked block was present on this machine, and is included
+   # in this message
+   | Present             ("p")
+   # DuplicateNotSent means the linked block was present on this machine, but I
+   # am not sending it (most likely duplicate)
+   # | DuplicateNotSent   ("d")
+   # Missing means I did not have the linked block, so I skipped over this part
+   # of the traversal
+   | Missing             ("m")
+   # DuplicateDAGSkipped means the DAG with this link points toward has already
+   # been traversed entirely in the course of this request
+   # so I am skipping over it entirely
+   # | DuplicateDAGSkipped ("s")
+} representation string
```

Comment:
Curious -- is there a reason we went with string? Given these are super short it's probably fine, just curious.

Reply:
A bias toward descriptiveness without costing bytes. On the Go side we get to use real strings in the enum (`"Present"` and `"Missing"`) to match the names here and in the message you have at least a small chance of _seeing_ what a message contains if you capture the raw data and decode the cbor. Same bias that's applied to selector's IPLD representation, I can look at `{"R":{"l":{"none":{}},":>":{"a":{">":{"@":{}}}}}}` and can work out what it is if I know enough, because the symbols are suggestive and more memorable than `{"0":{"1":{"none":{}},":>":{"2":{"3":{"4":{}}}}}}`

Reply:
> On the Go side we get to use real strings in the enum (`"Present"` and `"Missing"`) to match the names here

FYI, you could do that on bindnode with any IPLD Enum, no matter the representation strategy. It's just that for int reprs, you can *also* use an integer on the Go side, but you can still use a string like you do here.

Have to agree on the visual help part of the argument anyway, but wanted to clarify :)